### ISSUE TITLE
Add function calls

### ISSUE THREAD

**Issue by gombosg** [cequence-io/openai-scala-client#28]
Hey, are you working on this? It was just [released](https://openai.com/blog/function-calling-and-other-api-updates) and as I see it involves adding new request parameters.

If it's not planned, I can scratch my open-source itch and contribute PR sometime this week.

**Comment by peterbanda** [comment]
Yep. It has been released exactly when you posted this :) The version is `0.4.0`. Enjoy!
